packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -190,6 +191,38 @@ class StripeHelper {
     }
   }
 
+  /**
+   * Fetch a customer record from Stripe by id and return its userid metadata
+   * and the email.
+   *
+   * @param {Subscription} sub
+   * @returns {Promise<{uid: string, email: string} | {uid: undefined, email: undefined} | {uid: undefined, email: string}>}
+   */
+  async getCustomerUidEmailFromSubscription(sub) {
+    const customer = await this.stripe.customers.retrieve(
+      /** @type {string} */ (sub.customer)
+    );
+    if (customer.deleted) {
+      // Deleted customers lost their metadata so we can't send events for them
+      return { uid: undefined, email: undefined };
+    }
+    if (!(/** @type {Customer} */ (customer.metadata.userid))) {
+      Sentry.withScope(scope => {
```

Comment:
We should probably make a more concise helper for this as it seems to be propagating. Not needed in this commit though. 😄